### PR TITLE
Full CI integration for dependents' project

### PR DESCRIPTION
**Pull request by joao-paulo-parity**
related to https://github.com/paritytech/ci_cd/issues/234

related to https://github.com/paritytech/polkadot/pull/5713

:exclamation: should be merged together with https://github.com/paritytech/pipeline-scripts/pull/70

### CODE REVIEW

**Comment by joao-paulo-parity** [comment]
/cmd queue -c test-bench-bot $ pallet statemine pallet_balances

**Comment by joao-paulo-parity** [comment]
/cmd queue -c bench-bot $ pallet statemine pallet_balances

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -44,12 +45,20 @@ variables:
...
+    - if: $CI_COMMIT_REF_NAME =~ /^[0-9]+$/                         # PRs
+    - if: $CI_COMMIT_REF_NAME =~ /^cbs-.*/                          # Patched companion branches
+
 .common-refs:                      &common-refs
   # these jobs run always*
   rules:
     - if: $CI_PIPELINE_SOURCE == "schedule"
     - if: $CI_COMMIT_REF_NAME == "master"
-    - if: $CI_COMMIT_REF_NAME =~ /^[0-9]+$/                         # PRs
+    - !reference [.pr-refs, rules]
```
Comment (alvicsam) [inline .gitlab-ci.yml:61]:
```suggestion
    - if: $CI_COMMIT_REF_NAME =~ /^[0-9]+$/                         # PRs
    - if: $CI_COMMIT_REF_NAME =~ /^cbs-.*/                          # Patched companion branches
```

I would prefer not to mix styles

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -73,15 +82,37 @@ variables:
...
+  before_script:
+    - *check-passed-job
   tags:
     - kubernetes-parity-build
 
+#### stage:                        prepare
+
+prepare:
+  stage:                           prepare
+  <<:                              *kubernetes-env
```
Comment (alvicsam) [inline .gitlab-ci.yml:101]:
```suggestion
  <<:                              *docker-env
```
Should be executed on more powerful ci runners

**Review by alvicsam — CHANGES_REQUESTED** [review]

File: .gitlab-ci.yml (hunk 2 above)
Comment (joao-paulo-parity) [reply .gitlab-ci.yml:101]: Same response as https://github.com/paritytech/substrate/pull/11765#discussion_r931949160

File: .gitlab-ci.yml (hunk 1 above)
Comment (joao-paulo-parity) [reply .gitlab-ci.yml]: eb86bce408e6e793c80ff3e3d706b198b3dbc9e1

**Pull request closed without merging**